utils/misc.mli

```diff
@@ -450,7 +450,8 @@ val spellcheck : string list -> string -> string list
     [name] that it may be a typo for one of them. *)
 
 val did_you_mean :
-    Format_doc.formatter -> (unit -> string list) -> unit
+    ?pp:string Format_doc.printer -> Format_doc.formatter ->
+    (unit -> string list) -> unit
```

Comment:
If you parametrize over the printing, I would expect `string` to become a polymorphic parameter `'a` in the signature.

Reply:
This is a possibility but this requires to make the argument non-optional and currently there is only one meaningful use of the argument over seven calls to `did_you_mean`.